src/runtime/synchronization_common.h

```diff
@@ -1138,4 +1138,36 @@ WEAK void halide_cond_wait(struct halide_cond *cond, struct halide_mutex *mutex)
    fast_cond->wait(fast_mutex);
 }
 
+WEAK halide_mutex_array* halide_mutex_array_create(int sz) {
+    halide_mutex_array *array = (halide_mutex_array*)halide_default_malloc(
+        NULL, sizeof(halide_mutex_array));
+    if (array == NULL) {
+        return NULL;
+    }
+    array->array = (halide_mutex*)halide_default_malloc(
+        NULL, sz * sizeof(halide_mutex));
+    if (array->array == NULL) {
```

Comment:
halide_default_free(NULL, NULL) doesn't seem worth calling.

Reply:
Also, should probably use halide_malloc/halide_free, not the default implementation. These allocations could be large, and users might want them to come from a custom allocator.

Reply:
Done. I meant to call halide_free(NULL, array) so that was a bug.